src/app/common/loading-modal/loading-dialog.service.ts

```diff
@@ -0,0 +1,22 @@
+import {Injectable} from '@angular/core';
+import {MatSnackBar, MatSnackBarRef} from '@angular/material/snack-bar';
+import {LoadingModalComponent} from './loading-modal.component';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class LoadingDialogService {
+
+  public constructor(private snackBar: MatSnackBar) {
+  }
+
+  private snackBarRef!: MatSnackBarRef<LoadingModalComponent>;
+
+  public startLoading(message: string): void {
```

Comment:
I think you can find a better name for those two methods. You are not "starting" something, you are just showing loading indicator and hiding it :)